src/rgw/motr/sync/motr_sync_impl.cc

```diff
@@ -29,46 +47,65 @@ int MotrLock::lock(const std::string& lock_name,
   }
   int rc = 0;
   motr_lock_info_t lock_obj;
-  rc = _lock_provider->read_lock(lock_name, &lock_obj);
-  if (rc != 0 && rc != -ENOENT)
-    return rc;
+  // First, try to write lock object
+  struct motr_locker_info_t locker;
+  std::string s_locker_id = locker_id;
+  if (s_locker_id.empty()) {
+    s_locker_id = random_string(32);
```

Comment:
This random string can be calculated in the initialize method & can be used later on. In my view, there is no need to derive it for every call to "lock" method.

Reply:
@sachinpunadikar : This random string is required for every call to lock() to make sure the same client/caller has the possession of lock object after the lock is acquired by caller.

Reply:
@sachinpunadikar : Will invoke random_string() in the GC thread init (one time initlaization)

Reply:
Please note that the invocation of "random_string()" in the GC thread init (one time initialization) will be done in separate PR